### PR TITLE
deeper arch, command-line args

### PR DESCRIPTION
**Pull request by mudit1990**
Deeper architecture for model2, model3
command-line args for word corpus, email emb, user emb, hidden_dims 

can use these configurations to run in local
"dundundun" "Model2Deeper" "5" "150" "[256,256,256]" "6B" "100" "200"

### CODE REVIEW

File: code/model3deeper.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        # first hidden layer, linear layer with weights 2*constants.USER_EMB_SIZEx500
+        # this should be the size of <sender+receiver representation>
+        inp_dim = 2 * constants.USER_EMB_SIZE
+        self.hidden_layers, out_dim = self.create_hidden_layers(inp_dim, hidden_dims, use_batchnorm)
+        # final linear layer that outputs the predicted email representation
+        self.email_layer = nn.Linear(out_dim, constants.EMAIL_EMB_SIZE)
+        if pre_trained:
+            self.load(load_from)
+
+    def create_hidden_layers(self, inp_dim, hidden_dims, use_batchnorm):
```
Comment (mohit-surana) [inline code/model3deeper.py]: Can we pull this out as a common helper function?

File: code/model3.py [hunk 2]
```diff
@@ -22,7 +22,7 @@ class Model3(nn.Module, Model):
      representation is obtained by averaging embeddings from pre-trained word2vec model
     """
 
-    def __init__(self, pre_trained=False, load_from=None):
+    def __init__(self, pre_trained=False, load_from=None, hidden_dims=[500]):
```
Comment (mohit-surana) [inline code/model3.py:25]: Once model3deeper starts working, can it replace model3?
The hidden dims variable is not being used

**Review by mohit-surana — APPROVED** [review]
Looks good

File: code/model3.py (hunk 2 above)
Comment (mudit1990) [reply code/model3.py:25]: yes it can

File: code/model3deeper.py (hunk 1 above)
Comment (mudit1990) [reply code/model3deeper.py]: yep it can be moved to models.py ... i'll move it

**Pull request merged**